src/vector/backends/object.py

```diff
@@ -3091,6 +3087,11 @@ def obj(**coordinates: float) -> VectorObject:
     """
     is_momentum = False
     generic_coordinates = {}
+
+    for _, value in coordinates.items():
+        if not isinstance(value, (int, float)) or isinstance(value, bool):
+            raise TypeError("a coordinate must be of the type int or float")
```

Comment:
```suggestion
        if not isinstance(value, numbers.Real) or isinstance(value, bool):
            raise TypeError("a coordinate must be of the type int or float")
```

and `import numbers` at the top of the file. ([See docs](https://docs.python.org/3/library/numbers.html).)